src/spec.rs

```diff
@@ -47,23 +50,33 @@ use crate::boehm;
 /// and Z contiguously (provided size and alignment constraints are met).
 /// Conservative allocation is currently the only safe way to allocate such
 /// blocks.
-pub(crate) struct AllocationSpecializer<T> {
```

Comment:
A question: how were you able to get rid of the type parameter?

It's up to you but I wondered whether you might prefer something like:

```
#[cfg(not(test))] struct AllocationSpecializer;
#[cfg(test)] struct AllocationSpecializer { num_atomic: ..., ..., }
```

to make things clearer? It's perhaps as broad as it's long however!
struct AllocationSpecializer

Reply:
How about something like this: 18b60344b8cee4472d98dfec59f918f1670c57f6? 

> A question: how were you able to get rid of the type parameter?

It's still there, but instead of being constrained to the `AllocationSpecializer` struct, it exists on the `maybe_optimized_alloc::<T>` method. It doesn't make a huge difference really, especially as we construct a new `AllocationSpecializer` on each allocation (which is free, as it's a ZST). This does make the unit tests a bit tidier though because you can do multiple allocations of different types with the same `AllocationSpecializer`.

Reply:
I like https://github.com/softdevteam/libgc_internal/commit/18b60344b8cee4472d98dfec59f918f1670c57f6: what are your thoughts on it though?

>This does make the unit tests a bit tidier though because you can do multiple allocations of different types with the same AllocationSpecializer.

Ah, makes sense.

Reply:
I think I prefer it. `cfg` attributes on each field is a bit dirty and obscures the fact that this should be a ZST in real-life. Shall we go with it?

Reply:
Cool, let's keep it!